docs/examples/101/databricks-all-in-one-template-for-vnet-injection/main.bicep

```diff
@@ -1,76 +1,181 @@
+@description('Specifies whether to deploy Azure Databricks workspace with secure cluster connectivity (SCC) enabled or not (No Public IP)')
+param disablePublicIp bool = false
+
+@description('Location for all resources.')
+param location string = resourceGroup().location
+
+@description('The name of the network security group to create.')
 param nsgName string = 'databricks-nsg'
-param vnetName string = 'databricks-vnet'
-param workspaceName string
-param privateSubnetName string = 'private-subnet'
-param publicSubnetName string = 'public-subnet'
 
 @allowed([
+  'trial'
   'standard'
   'premium'
 ])
+@description('The pricing tier of workspace.')
 param pricingTier string = 'premium'
 
-param location string = resourceGroup().location
-param vnetCidr string = '10.179.0.0/16'
+@description('Cidr range for the private subnet.')
 param privateSubnetCidr string = '10.179.0.0/18'
+
+@description('The name of the private subnet to create.')
+param privateSubnetName string = 'private-subnet'
+
+@description('Cidr range for the public subnet..')
 param publicSubnetCidr string = '10.179.64.0/18'
 
+@description('The name of the public subnet to create.')
+param publicSubnetName string = 'public-subnet'
+
+@description('Cidr range for the vnet.')
+param vnetCidr string = '10.179.0.0/16'
+
+@description('The name of the virtual network to create.')
+param vnetName string = 'databricks-vnet'
+
+@description('The name of the Azure Databricks workspace to create.')
+param workspaceName string
+
 var managedResourceGroupName = 'databricks-rg-${workspaceName}-${uniqueString(workspaceName, resourceGroup().id)}'
 var managedResourceGroupId = '${subscription().id}/resourceGroups/${managedResourceGroupName}'
 
-var publicSubnet = {
-  name: publicSubnetName
+resource nsg 'Microsoft.Network/networkSecurityGroups@2020-05-01' = {
```

Comment:
Quite a few changes from the original quickstart, and I don't see any obvious reason, so I kept mostly what was in the original quickstart (after decompile).